src/obu.rs

```diff
@@ -1302,48 +1301,59 @@ fn parse_cdef(
 
 fn parse_restoration(
     seqhdr: &Rav1dSequenceHeader,
-    all_lossless: c_int,
-    super_res_enabled: c_int,
-    allow_intrabc: c_int,
+    all_lossless: bool,
+    super_res_enabled: bool,
+    allow_intrabc: bool,
     debug: &Debug,
     gb: &mut GetBits,
 ) -> Rav1dFrameHeader_restoration {
     let r#type;
     let unit_size;
-    if (all_lossless == 0 || super_res_enabled != 0)
-        && seqhdr.restoration != 0
-        && allow_intrabc == 0
-    {
-        let type_0 = gb.get_bits(2) as Rav1dRestorationType;
+    if (!all_lossless || super_res_enabled) && seqhdr.restoration != 0 && !allow_intrabc {
+        let type_0 = Rav1dRestorationType::from_repr(gb.get_bits(2) as usize).unwrap();
         r#type = if seqhdr.monochrome == 0 {
             [
                 type_0,
-                gb.get_bits(2) as Rav1dRestorationType,
-                gb.get_bits(2) as Rav1dRestorationType,
+                Rav1dRestorationType::from_repr(gb.get_bits(2) as usize).unwrap(),
+                Rav1dRestorationType::from_repr(gb.get_bits(2) as usize).unwrap(),
             ]
         } else {
-            [type_0, RAV1D_RESTORATION_NONE, RAV1D_RESTORATION_NONE]
+            [
+                type_0,
+                Rav1dRestorationType::None,
+                Rav1dRestorationType::None,
+            ]
         };
 
-        unit_size = if r#type[0] != 0 || r#type[1] != 0 || r#type[2] != 0 {
-            // Log2 of the restoration unit size.
-            let mut unit_size_0 = 6 + seqhdr.sb128;
-            if gb.get_bit() {
-                unit_size_0 += 1;
-                if seqhdr.sb128 == 0 {
-                    unit_size_0 += gb.get_bit() as c_int;
-                }
+        unit_size = match r#type {
+            [Rav1dRestorationType::None, Rav1dRestorationType::None, Rav1dRestorationType::None] => {
+                [8, 0]
             }
-            let mut unit_size_1 = unit_size_0;
-            if (r#type[1] != 0 || r#type[2] != 0) && seqhdr.ss_hor == 1 && seqhdr.ss_ver == 1 {
-                unit_size_1 -= gb.get_bit() as c_int;
+            _ => {
+                // Log2 of the restoration unit size.
+                let mut unit_size_0 = 6 + seqhdr.sb128;
+                if gb.get_bit() {
+                    unit_size_0 += 1;
+                    if seqhdr.sb128 == 0 {
+                        unit_size_0 += gb.get_bit() as c_int;
+                    }
+                }
+
+                let unit_size_1 = if (r#type[1] != Rav1dRestorationType::None
+                    || r#type[2] != Rav1dRestorationType::None)
```

Comment:
```suggestion
                let unit_size_1 = if !(matches!(r#type, [_, Rav1dRestorationType::None, Rav1dRestorationType::None]))
```

Reply:
sadly your suggestion formats as

```rust
                let unit_size_1 = if !(matches!(
                    r#type,
                    [_, Rav1dRestorationType::None, Rav1dRestorationType::None]
                )) && seqhdr.ss_hor == 1
                    && seqhdr.ss_ver == 1
                {
                    unit_size_0 - gb.get_bit() as c_int
                } else {
                    unit_size_0
                };
```
which I think is less clear

Reply:
Oh yeah, it's fine as is then.  Maybe if `Rav1dRestorationType::None` were replaced with `Option`'s `None` it'd be better, but that seems like more work than it's worth.